Oswaldo Vazquez Aparicio/26 Agosto/miniproject/ShoppingList.java

```diff
@@ -0,0 +1,30 @@
+package miniproject;
+
+import java.util.LinkedList;
+
+public class ShoppingList {
+
+    private final LinkedList<Product> products;
```

Comment:
Why a LinkedList ?? why?
insert-"explica tu maqueta jovencito..."-GIF

Reply:
Well, when I started to make the exercise I have 3 options which were Array, ArrayList and LinkedList. I didn't choose ArrayList because it has an initial capacity that you have to provide or you can use the default initial capacity, and when you use more than the initial capacity backstage a new ArrayList is created copying the elements of the old one and making the capacity bigger but this with a lot of data can cause problems and I didn't choose
an Array because you have to provide the number of elements that the Array is gonna contain and you can't add more elements after at least you do the same than in an ArrayList, so the reason I decided to use a LinkedList is that the size is not needed to be provided because is dynamic and add elements is not expensive and iterate through the LinkedList have the same cost than the others two data structures and in this toy exercise I just adding and iterating through the elements, but thinking more about it. Probably, it would be better to use an Array because the elements are not gonna be modified in runtime, Oops!.

So I think I should rewrite it in this way

![Captura de pantalla 2021-08-28 123347](https://user-images.githubusercontent.com/47643347/131226116-f0b17ac5-acb0-451c-b837-2d39a8687f6d.png)  

![Captura de pantalla 2021-08-28 123358](https://user-images.githubusercontent.com/47643347/131226271-48a3596d-b9ec-4dc5-9e9a-4e85feae6fbe.png)

Reply:
Today is Collections lesson, let's talk about it later